### PR TITLE
makes paper stack burning more consistent

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
makes paper stack burning more consistent, uses ishot (any hot object) instead of checking for only lighters
You now always burn the entire stack
also fixes a bunch of runtimes
fixes https://github.com/ParadiseSS13/Paradise/issues/16218

## Why It's Good For The Game
This is just better, more consistent with other stuff and less jank to use

## Testing
Compiled, ran, burned some books

## Changelog
:cl:
tweak: paper stack burning is more consistent and no longer requires just a lighter
/:cl:

### CODE REVIEW

File: code/modules/paperwork/paper_bundle.dm [hunk 1]
```diff
@@ -77,29 +77,24 @@
...
-					user.unEquip(src)
+	if(!do_after(user, 2 SECONDS, target = src))
+		return
+	user.visible_message("[class][user] burns right through \the [src], turning it to ash. It flutters through the air before settling on the floor in a heap.</span>", \
+	"[class]You burn right through \the [src], turning it to ash. It flutters through the air before settling on the floor in a heap.</span>")
 
-				new /obj/effect/decal/cleanable/ash(get_turf(src))
-				qdel(src)
+	if(user.is_in_inactive_hand(src))
+		user.unEquip(src)
```
Comment (Contrabang) [inline code/modules/paperwork/paper_bundle.dm:94]: All of the cases I can think of, it should always be unequipped
```suggestion
	user.unEquip(src)
```

File: code/modules/paperwork/paper_bundle.dm [hunk 2]
```diff
@@ -77,29 +77,23 @@
...
+	user.visible_message("[class][user] holds [heating_object] up to [src], it looks like [user.p_theyre()] trying to burn it!</span>", "[class]You hold [heating_object] up to [src], burning it slowly.</span>")
 
-		spawn(20)
-			if(get_dist(src, user) < 2 && user.get_active_hand() == P && P.lit)
-				user.visible_message("[class][user] burns right through \the [src], turning it to ash. It flutters through the air before settling on the floor in a heap.</span>", \
-				"[class]You burn right through \the [src], turning it to ash. It flutters through the air before settling on the floor in a heap.</span>")
-
-				if(user.is_in_inactive_hand(src))
-					user.unEquip(src)
+	if(!do_after(user, 2 SECONDS, target = src) && is_hot(heating_object))
```
Comment (Contrabang) [inline code/modules/paperwork/paper_bundle.dm:88]: Sorry, I got the logic here wrong
```suggestion
	if(!do_after(user, 2 SECONDS, target = src) || !is_hot(heating_object))
```

File: code/modules/paperwork/paper_bundle.dm [hunk 3]
```diff
@@ -77,29 +77,23 @@
...
+		return
+	user.visible_message("[class][user] burns right through \the [src], turning it to ash. It flutters through the air before settling on the floor in a heap.</span>", \
+	"[class]You burn right through \the [src], turning it to ash. It flutters through the air before settling on the floor in a heap.</span>")
```
Comment (lewcc) [inline code/modules/paperwork/paper_bundle.dm:91]:
```suggestion
	user.visible_message("[class][user] burns right through [src], turning it to ash. It flutters through the air before settling on the floor in a heap.</span>", \
	"[class]You burn right through [src], turning it to ash. It flutters through the air before settling on the floor in a heap.</span>")
```

File: code/modules/paperwork/paper_bundle.dm [hunk 4]
```diff
@@ -77,29 +77,23 @@
...
 
-	if(P.lit && !user.restrained())
-		if(istype(P, /obj/item/lighter/zippo))
-			class = "<span class='rose'>"
+	if(istype(heating_object, /obj/item/lighter/zippo))
+		class = "<span class='rose'>"
 
-		user.visible_message("[class][user] holds [P] up to [src], it looks like [user.p_theyre()] trying to burn it!</span>", \
-		"[class]You hold [P] up to [src], burning it slowly.</span>")
+	user.visible_message("[class][user] holds [heating_object] up to [src], it looks like [user.p_theyre()] trying to burn it!</span>", "[class]You hold [heating_object] up to [src], burning it slowly.</span>")
```
Comment (lewcc) [inline code/modules/paperwork/paper_bundle.dm:86]: Can you move the spans themselves into the visible messages, and just pull out the actual classes themselves (e.g. warning, rose)

**Comment by DGamerL** [comment]
Could you link #22995 so it closes correctly?

**Comment by DGamerL** [comment]
Please link #16218 as it'll be fixed by this PR!

**Review by DGamerL — CHANGES_REQUESTED** [review]
Please adress Lewc's review, this is going stale

File: code/modules/paperwork/paper_bundle.dm (hunk 4 above)
Comment (GDNgit) [reply code/modules/paperwork/paper_bundle.dm]: reviews addressed

**Review by DGamerL — APPROVED** [review]
Please link #16218 as it'll be fixed by this PR!

**Review by Contrabang — APPROVED** [review]

**Pull request merged**